### PR TITLE
Feature/163 permission granting prompt

### LINKED ISSUE #160: Push Permissions
## things to clarify
- [x] (on IOS) when should we ask the user to allow permission, on the first app start?
- [x] if the user declines it should we ask him to enable it again when he/she tries to change notification settings?
- [x] should we show the user the warning for enabling their push notification on both platforms?

_Originally posted by @franjs in https://github.com/litentry/litentry-app/pull/156#discussion_r657979679_

### LINKED ISSUE #163: Milestone 2 | Permission granting prompt
> As an iOS user I want to be asked more than once if I want to enable push notifications.

iOS requires its user to grant permission for an app to send them push notifications. The native permission can only be shown once. To give the opportunity to the user to be asked multiple times we need a notification permission prompt. 

*Trigger: When the user open the app for the very first time, then every ... (to be completed) 
UI : Bell-outline (Eva Icons) 
image

*Copy:
          - Header: "Turn on Notifications"
          - Body: "Enable notifications to make sure you don't miss out on important events"
          - 1st Button : "Allow notifications"
          - 2nd Button: "Skip"

* If the User clicks on "Allow notifications", we show him the iOS native permission for push notification
* If he clicks on "Skip", he lands on the home screen

### PR DESCRIPTION
**Pull request by ahoseinian**
### Description
- show the prompt screen when the user open the app for the very first time
- ask users to allow permission or ask them to give the permissions from settings app.

fixes #163

### CODE REVIEW

File: src/screen/PermissionGrantingPrompt.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+function permissionAllowed(status: FirebaseMessagingTypes.AuthorizationStatus) {
+  return status === messaging.AuthorizationStatus.AUTHORIZED || status === messaging.AuthorizationStatus.PROVISIONAL;
+}
+
+const PERMISSION_GRANTING_SCREEN_SHOWN = 'PERMISSION_GRANTING_SCREEN_SHOWN';
+
+export function useShowPushPermissionScreen() {
+  return useQuery('permission_granted', async () => {
+    return (
+      Platform.OS === 'ios' &&
```
Comment (geastwood) [inline src/screen/PermissionGrantingPrompt.tsx:85]: Can we manage this `permission` all with the help of `FirebaseMessaging`?

Comment (ahoseinian) [reply src/screen/PermissionGrantingPrompt.tsx:85]: yes as far as I checked it gives us all the functions we need.

**Comment by ahoseinian** [comment]
@franjs can you take a look at this also?

File: src/screen/PermissionGrantingPrompt.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+});
+
+function permissionAllowed(status: FirebaseMessagingTypes.AuthorizationStatus) {
+  return status === messaging.AuthorizationStatus.AUTHORIZED || status === messaging.AuthorizationStatus.PROVISIONAL;
+}
+
+const PERMISSION_GRANTING_SCREEN_SHOWN = 'PERMISSION_GRANTING_SCREEN_SHOWN';
+
+export function useShowPushPermissionScreen() {
+  return useQuery('permission_granted', async () => {
```
Comment (silva-fj) [inline src/screen/PermissionGrantingPrompt.tsx:83]: Do we need to cache this?

**Review by silva-fj — APPROVED** [review]

File: src/screen/PermissionGrantingPrompt.tsx (hunk 2 above)
Comment (ahoseinian) [reply src/screen/PermissionGrantingPrompt.tsx:83]: not necessary I just used this as a quick way to have `isLoading` on async function. I don't think it hurts though.

**Pull request merged**